### PR TITLE
Performance Improvements for Indexing

### PR DESCRIPTION
**Pull request by sleidig**
see issue: #462

### Visible/Frontend Changes
- [x] extended database sync indicator to provide details on other background processes like indexing

 UI: indexing-indicators

### Architectural/Backend Changes
- [x] Performance tests to benchmark indexing (results detailed in #462)
- [x] Creating and querying indices is now done through the new `DatabaseIndexingService` instead of directly on `Database`
- [x] Trigger initial indexing immediately after login (together with creation of the index) to prepare database for faster queries by user

### CODE REVIEW

File: src/app/utils/performance-tests.spec.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 150000;
+  }));
+
+  it("sync initial and indexing", async () => {
+    // delete previously synced database; uncomment this to start with a clean state and test an initial sync.
+    // await deleteAllIndexedDB(db => true);
+
+    const session = TestBed.inject<SessionService>(SessionService);
+    const syncTimer = new Timer(true);
+    await session.login("demo", "pass");
```
Comment (TheSlimvReal) [inline src/app/utils/performance-tests.spec.ts:63]: is it okay to have credentials as plaintext here?

**Review by TheSlimvReal — APPROVED** [review]
Looks very good, just a few small notes.

File: src/app/utils/performance-tests.spec.ts (hunk 1 above)
Comment (sleidig) [reply src/app/utils/performance-tests.spec.ts:63]: good point. even though the credentials are trivial this shouldn't be in the code base

Comment (sleidig) [reply src/app/utils/performance-tests.spec.ts:63]: a really clean solution would probably be an .env file to set environment variables and put that file on .gitignore. That way we could even easily configure these in the CI as environment variables.

But for now, I am taking the shortcut on just putting variables with placeholders to finally get this PR out of the door.

**Pull request merged**